mesonbuild/mesonmain.py

```diff
@@ -234,6 +234,25 @@ def set_meson_command(mainfile: str) -> None:
     from . import mesonlib
     mesonlib.set_meson_command(mainfile)
 
+def validate_original_args(args):
+    import mesonbuild.options
+    import itertools
```

Comment:
There is no reason for `itertools` to be imported here instead of at the top level, it's just a code smell.

Reply:
Whether it is a *good* reason or not is up for debate, but you're 100% wrong to say that there's "no" reason. The reason is described at the top of the file, right where you suggest moving the import to. Is it load-bearing in this particular instance? I don't know.

I'm pretty sure there's a CI test that will fail if the import is moved to the top level.